File: fix_layout.py
import re
import os

filepath = r"d:\HEALTHTRACKESNESABA\index.html"
with open(filepath, "r", encoding="utf-8") as f:
    html = f.read()

# 1. Fix the top brand logo and text
old_brand = """<div class="brand">
        <div class="brand-logo">HT</div>
        <div>
          <h1>HEALTH TRACK ESNESABA</h1>
          <p>Program 30 Hari Hidup Sehat Bersama ? SMP NEGERI 1 BANYUBIRU</p>
        </div>
      </div>"""
new_brand = """<div class="brand">
        <img src="logo.png" alt="Logo Sekolah" style="width: 50px; height: 50px; border-radius: 12px; object-fit: contain;">
        <div>
          <h1>HEALTH TRACK ESNESABA</h1>
          <p>Program 30 Hari Hidup Sehat Bersama<br>SMP NEGERI 1 BANYUBIRU</p>
        </div>
      </div>"""
if old_brand in html:
    html = html.replace(old_brand, new_brand)
else:
    # try regex for flexible spacing
    html = re.sub(r'<div class="brand-logo">HT</div>\s*<div>\s*<h1>HEALTH TRACK ESNESABA</h1>\s*<p>Program 30 Hari Hidup Sehat Bersama \? SMP NEGERI 1 BANYUBIRU</p>\s*</div>',
                  '<img src="logo.png" alt="Logo Sekolah" style="width: 50px; height: 50px; border-radius: 12px; object-fit: contain;">\n        <div>\n          <h1>HEALTH TRACK ESNESABA</h1>\n          <p>Program 30 Hari Hidup Sehat Bersama<br>SMP NEGERI 1 BANYUBIRU</p>\n        </div>', html)

# 2. Fix the login quotes! The user said: "AUTHENTICATION REQUIRED GANTI DENGAN QUOTE MOTIVASI BAHASA INDONESIA, QUOTE OTOMATIS BERUBAH KETIKA HALAMAN DI REFRESH"
# The previous agent had BOTH `loginQuote` AND `authQuote` active.
# Let's remove the loginQuote element entirely, as they probably want the quote where AUTHENTICATION REQUIRED was!
html = re.sub(r'<div class="hero-emoji" id="loginQuote" [^>]+></div>\s*', '', html)

# 3. Make sure GURU is in the classCode dropdown!
# The classCode dropdown ends with: <option>9F</option>\n          </select>
classcode_regex = r'(<select id="classCode">[\s\S]*?<option>9F</option>)\s*</select>'
if 'GURU' not in re.search(classcode_regex, html).group(0):
    html = re.sub(classcode_regex, r'\1\n            <option>GURU</option>\n          </select>', html)

# 4. Fix any lingering `?` characters in other strings if any exist where they shouldn't.
# The user might have seen `?` in top stats.
html = html.replace('?"', '-"')
html = html.replace("? ", "- ")

# 5. Fix the double-rendering of the initNutrifoodRain() logic
# If `initNutrifoodRain();` appears twice due to earlier replacements, remove the second one.
# But it only appeared once when I checked. Let's make sure.

with open(filepath, "w", encoding="utf-8") as f:
    f.write(html)
print("Fixes applied.")

File: fix_layout.js
const fs = require('fs');
const filepath = 'd:\\HEALTHTRACKESNESABA\\index.html';
let html = fs.readFileSync(filepath, 'utf8');

// 1. Fix the top brand logo and text
const old_brand = `<div class="brand">
        <div class="brand-logo">HT</div>
        <div>
          <h1>HEALTH TRACK ESNESABA</h1>
          <p>Program 30 Hari Hidup Sehat Bersama ? SMP NEGERI 1 BANYUBIRU</p>
        </div>
      </div>`;
const new_brand = `<div class="brand">
        <img src="logo.png" alt="Logo Sekolah" style="width: 50px; height: 50px; border-radius: 12px; object-fit: contain;">
        <div>
          <h1>HEALTH TRACK ESNESABA</h1>
          <p>Program 30 Hari Hidup Sehat Bersama<br>SMP NEGERI 1 BANYUBIRU</p>
        </div>
      </div>`;
if (html.includes(old_brand)) {
    html = html.replace(old_brand, new_brand);
} else {
    // regex fallback
    html = html.replace(/<div class="brand-logo">HT<\/div>\s*<div>\s*<h1>HEALTH TRACK ESNESABA<\/h1>\s*<p>Program 30 Hari Hidup Sehat Bersama \? SMP NEGERI 1 BANYUBIRU<\/p>\s*<\/div>/,
                  `<img src="logo.png" alt="Logo Sekolah" style="width: 50px; height: 50px; border-radius: 12px; object-fit: contain;">
        <div>
          <h1>HEALTH TRACK ESNESABA</h1>
          <p>Program 30 Hari Hidup Sehat Bersama<br>SMP NEGERI 1 BANYUBIRU</p>
        </div>`);
}

// 2. Remove the first hero-emoji loginQuote div so it doesn't show up twice
html = html.replace(/<div class="hero-emoji" id="loginQuote" [^>]+><\/div>\s*/, '');

// 3. Make sure GURU is in the classCode dropdown!
const classCodeRegex = /(<select id="classCode">[\s\S]*?<option>9F<\/option>)\s*<\/select>/;
const match = html.match(classCodeRegex);
if (match && !match[0].includes('GURU')) {
    html = html.replace(classCodeRegex, `$1\n            <option>GURU</option>\n          </select>`);
}

// 4. Fix any lingering `?` characters
html = html.replace(/\?"/g, '-"');
html = html.replace(/\? /g, '- ');

fs.writeFileSync(filepath, html, 'utf8');
console.log("Node fix script completed.");
